src/compas-editors/autogen-substation.ts

```diff
@@ -342,12 +346,9 @@ export default class CompasAutogenerateSubstation extends LitElement {
    * @returns filtered content
    */
   extractNames(content: string[]) {
-    //Check for empty elements
-    const elementNames = content.filter(Boolean);
-
-    //return list of names after filtering out the duplicate elements
-    return elementNames.filter(
-      (value, index) => value && elementNames.indexOf(<string>value) === index
+    //return list of names after filtering out the empty and duplicate elements
+    return content.filter(
+      (value, index) => value && content.indexOf(<string>value) === index
```

Comment:
value is always a string, no need to cast it to a string.

Reply:
Thanks, Ill change it